main.tf

```diff
@@ -152,8 +152,19 @@ resource "azurerm_kubernetes_cluster" "main" {
   oidc_issuer_enabled = var.oidc_issuer_enabled
 
   tags = var.tags
-}
 
+  lifecycle {
+    precondition {
+      condition     = (var.client_id != "" && var.client_secret != "") || (var.identity_type != "")
```

Comment:
identity_type should have `nullable = false` in the variable definition ?

Reply:
`identity_type` has a validation block, `null` won't pass the validation so I think it's ok for this case.